cmd/kubelet/app/server.go

```diff
@@ -285,6 +287,73 @@ HTTP server: The kubelet can also listen for HTTP and respond to a simple API
 	return cmd
 }
 
+// loadKubeletConfiguration serializes an YAML in the internal KubeletConfiguration
+func loadKubeletConfiguration(configFile string, args []string) (*kubeletconfiginternal.KubeletConfiguration, error) {
+	kubeletConfig, err := loadConfigFile(configFile)
+	if err != nil {
+		klog.Fatal(err)
+	}
+
+	// We must enforce flag precedence by re-parsing the command line into the new object.
+	// This is necessary to preserve backwards-compatibility across binary upgrades.
+	// See issue #56171 for more details.
+	if err := kubeletConfigFlagPrecedence(kubeletConfig, args); err != nil {
+		return nil, err
+	}
+
+	return kubeletConfig, nil
+}
+
+// loadKubeletInstanceConfiguration serializes the instance configuration in the internal KubeletInstanceConfiguration
+func loadKubeletInstanceConfiguration(
+	kubeletConfig *kubeletconfiginternal.KubeletConfiguration, instanceConfigFile string,
+) (*kubeletconfiginternal.KubeletInstanceConfiguration, error) {
+
+	// A default external KubeletInstanceConfiguration object initialized with the shared KubeletConfiguration fields.
+	defaultInstance := &kubeletconfigv1beta1.KubeletInstanceConfiguration{
+		RegisterNode:     &kubeletConfig.RegisterNode,
+		HostnameOverride: kubeletConfig.HostnameOverride,
+		NodeIP:           kubeletConfig.NodeIP,
+	}
+
```

Comment:
The default instance holds an initial external type with the shared internal type copied over, it will be used later for the instance config decode on top of it.
A point of refactoring here is the way these structs are being merged.